ksql-engine/src/main/java/io/confluent/ksql/function/udf/array/GenerateSeries.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.array;
+
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This UDF constructs an array containing an array of INTs or BIGINTs in the specified range
+ */
+@UdfDescription(name = "GENERATE_SERIES", description = "Construct an array of a range of values")
+public class GenerateSeries {
+
+  @Udf
+  public List<Integer> generateSeriesInt(
+      @UdfParameter(description = "The beginning of the series") final int startInclusive,
+      @UdfParameter(description = "Marks the end of the series (exclusive)") final int endExclusive
```

Comment:
Postgres seems to be inclusive of the end index (https://www.postgresql.org/docs/9.1/functions-srf.html). This pains me to no end, but we should make sure to be consistent so we don't need to do what we did with `SUBSTRING` - again would love comment about these functions from @blueedgenick

Reply:
If it's my 2c.. I'd say we shouldn't slavishly copy what existing DBs do. Many of the reasons they do things in a certain way are because the decisions were made long ago, and I'm sure they would want to change them if they could, but they can't for compatibility reasons. But it's different for us, we're new and we have an opportunity to do things better. Also.. if we're honest most of the users of KSQL will be developers not DBAs and exclusive upper ranges are more common in most programming languages :)

Reply:
I think there's a few compelling benefits from "slavishly copying what existing DBs do":

1. we can iterate faster by providing a framework for resolving decisions that are mostly a matter of opinion
2. we stay familiar to engineers that have been using SQL systems which are not necessarily developers (not a small target audience)
3. we position ourselves as a SQL compatible system (this one is a stretch). I might want to run the same queries on my postgresDB as on my KSQL cluster (esp. when we talk about pull queries)

Obviously, if there's strong objective reason to differ then we shouldn't rule out discussion - but I think the bar should be pretty high to diverge from cannon.

Reply:
I can change it to be inclusive. I'd rather just make the change than have a long discussion about it, as code freeze is on us soon :)

Reply:
I've made the changes, but I look forward to several years of answering user group questions from disgruntled engineers about why we chose upper bound to be inclusive ;)